### PR TITLE
Drone data class and improvments in the drone part

### PR DESCRIPTION
**Pull request by GabrielFleischer**
At first I was only porting the DroneData class that hold drone useful drone data as LiveDatas. But I also setup Hilt and fixed some tests.

### CODE REVIEW

**Review by Anghul — APPROVED** [review]
If you have time, can you add a way to access the uri of the video stream (can be found in camera.videoStrimSettings, https://mavsdk.mavlink.io/develop/en/api_reference/structmavsdk_1_1_camera_1_1_video_stream_settings.html)
If you don't, I can do it myself directly on my branch once yours is merged with the main

File: app/build.gradle [hunk 1]
```diff
@@ -68,8 +68,8 @@ dependencies {
     implementation 'com.mapbox.mapboxsdk:mapbox-android-plugin-annotation-v9:0.8.0'
 
     // Import mavsdk to control the drone
-    implementation 'io.mavsdk:mavsdk:0.3.1'
-    implementation 'io.mavsdk:mavsdk-server:0.3.1'
+    implementation 'io.mavsdk:mavsdk:0.8.0'
+    implementation 'io.mavsdk:mavsdk-server:0.8.0'
```
Comment (Squalene) [inline app/build.gradle:72]: Jonas Vauterin told me that they solved a bug in mavsdk-server:0.8.0 and now mavsdk-server:0.8.1 is the one most up to date.

**Review by Squalene — CHANGES_REQUESTED** [review]
Great work, will they be a way for the mapping mission builder to have access to the camera parameters such as https://github.com/mavlink/MAVSDK-Proto/blob/main/protos/camera/camera.proto#L367-L376?

File: app/build.gradle (hunk 1 above)
Comment (GabrielFleischer) [reply app/build.gradle:72]: I bumped the version.
And for the camera parameters, I added a camera resolution field, but if you want something more, I can add it very easily

**Review by Squalene — APPROVED** [review]

**Pull request merged**